records/labels/labels.go

```diff
@@ -0,0 +1,83 @@
+package labels
+
+import (
+	"bytes"
+	"strings"
+)
+
+// Sep is the default domain fragment separator.
+const Sep = "."
+
+// DomainFrag mangles the given name in order to produce a valid domain fragment.
+// A valid domain fragment will consist of one or more host name labels
+// concatenated by the given separator.
+func DomainFrag(name, sep string, label Func) string {
+	var labels []string
+	for _, part := range strings.Split(name, sep) {
+		if lab := label(part); lab != "" {
+			labels = append(labels, lab)
```

Comment:
**1st step:** This line causes a lot of mallocs by growing the slice. We can easily pre-allocate a `labels` array big enough for any sensible domain name.
**2nd step:** why collect the labels in an array at all? Join them in place => no malloc for the labels slice at all.